lib/wallet/src/Cardano/Wallet/DB.hs

```diff
@@ -507,14 +516,25 @@ mkDBLayerFromParts ti DBLayerCollection{..} = DBLayer
                     . filterMinWithdrawal minWithdrawal
                     $ inLedgers <> filter (
                             (||)
-                            <$> hasStatus WtxMeta.Pending
+                            <$> hasStatus WTxMeta.Pending
                             <*> hasStatus WTxMeta.Expired
                             ) inSubmissions
             Nothing ->
                 pure []
     , getTx = \wid txid -> wrapNoSuchWallet wid $ do
-        Just tip <- readCurrentTip wid -- wallet exists
-        (getTx_ dbTxHistory) wid txid tip
+        readCurrentTip wid >>= \case
+            Just tip -> do
+                historical <- getTx_ dbTxHistory wid txid tip
+                case historical of
+                    Just tx -> pure $ Just tx
+                    Nothing -> do
+                        inSubmission <-
+                            getInSubmissionTransaction_ dbPendingTxs wid txid
```

Comment:
Wait. What about the `Expired` transactions? We need to include them here — are they included in `getInSubmissionTransaction_`? 🤔 I think so, but the name and document seem to suggest otherwise?

Reply:
InSubmission is all transactions in submissions unfiltered

Reply:
I fixed the documentation about that and in other places